### DISCUSSION TITLE
无法安装paddleclas

### DISCUSSION THREAD

**Discussion by zhh8689** [PaddlePaddle/PaddleOCR#12624] [Q&A]
系统环境/System Environment：苹果 m2

版本号/Version：Paddle： 2.5.2 PaddleOCR： 2.7.0.3 paddlenlp: 2.6.1  SWIG:  4.1.1 faiss-cpu: 1.7.4

运行指令/Command Code：

pip3 install "paddleclas>=2.4.3" -i https://pypi.tuna.tsinghua.edu.cn/simple

完整报错/Complete Error Message：
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting paddleclas>=2.4.3
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/1f/d5/2544711d7bf23947f36bcfce81447af750adebb92f135fcaa28dfc014929/paddleclas-2.5.1-py3-none-any.whl (342 kB)
Requirement already satisfied: opencv-python in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from paddleclas>=2.4.3) (4.6.0.66)
Requirement already satisfied: tqdm in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from paddleclas>=2.4.3) (4.64.1)
Requirement already satisfied: PyYAML>=5.1 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from paddleclas>=2.4.3) (6.0)
Collecting prettytable
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/4d/81/316b6a55a0d1f327d04cc7b0ba9d04058cb62de6c3a4d4b0df280cbe3b0b/prettytable-3.9.0-py3-none-any.whl (27 kB)
Collecting easydict
  Using cached easydict-1.11-py3-none-any.whl
Requirement already satisfied: visualdl>=2.2.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from paddleclas>=2.4.3) (2.5.3)
Collecting faiss-cpu==1.7.1.post2
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/87/91/bf8ea0d42733cbb04f98d3bf27808e4919ceb5ec71102e21119398a97237/faiss-cpu-1.7.1.post2.tar.gz (41 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: ujson in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from paddleclas>=2.4.3) (5.4.0)
Collecting gast==0.3.3
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/d6/84/759f5dd23fec8ba71952d97bcc7e2c9d7d63bdc582421f3cd4be845f0c98/gast-0.3.3-py2.py3-none-any.whl (9.7 kB)
Requirement already satisfied: scipy>=1.0.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from paddleclas>=2.4.3) (1.10.1)
Requirement already satisfied: pillow in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from paddleclas>=2.4.3) (9.5.0)
Requirement already satisfied: scikit-learn>=0.21.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from paddleclas>=2.4.3) (1.2.1)
Requirement already satisfied: threadpoolctl>=2.0.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from scikit-learn>=0.21.0->paddleclas>=2.4.3) (2.2.0)
Requirement already satisfied: joblib>=1.1.1 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from scikit-learn>=0.21.0->paddleclas>=2.4.3) (1.1.1)
Requirement already satisfied: numpy>=1.17.3 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from scikit-learn>=0.21.0->paddleclas>=2.4.3) (1.23.5)
Requirement already satisfied: rarfile in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (4.1)
Requirement already satisfied: psutil in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (5.9.5)
Requirement already satisfied: packaging in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (22.0)
Requirement already satisfied: flask>=1.1.1 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (2.2.2)
Requirement already satisfied: bce-python-sdk in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (0.8.92)
Requirement already satisfied: six>=1.14.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (1.16.0)
Requirement already satisfied: protobuf>=3.20.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (3.20.2)
Requirement already satisfied: Flask-Babel>=3.0.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (4.0.0)
Requirement already satisfied: matplotlib in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (3.7.0)
Requirement already satisfied: requests in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (2.31.0)
Requirement already satisfied: pandas in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from visualdl>=2.2.0->paddleclas>=2.4.3) (1.5.3)
Requirement already satisfied: wcwidth in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from prettytable->paddleclas>=2.4.3) (0.2.5)
Requirement already satisfied: Werkzeug>=2.2.2 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from flask>=1.1.1->visualdl>=2.2.0->paddleclas>=2.4.3) (2.2.2)
Requirement already satisfied: itsdangerous>=2.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from flask>=1.1.1->visualdl>=2.2.0->paddleclas>=2.4.3) (2.0.1)
Requirement already satisfied: Jinja2>=3.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from flask>=1.1.1->visualdl>=2.2.0->paddleclas>=2.4.3) (3.1.2)
Requirement already satisfied: click>=8.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from flask>=1.1.1->visualdl>=2.2.0->paddleclas>=2.4.3) (8.1.7)
Requirement already satisfied: Babel>=2.12 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from Flask-Babel>=3.0.0->visualdl>=2.2.0->paddleclas>=2.4.3) (2.12.1)
Requirement already satisfied: pytz>=2022.7 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from Flask-Babel>=3.0.0->visualdl>=2.2.0->paddleclas>=2.4.3) (2022.7)
Requirement already satisfied: future>=0.6.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from bce-python-sdk->visualdl>=2.2.0->paddleclas>=2.4.3) (0.18.3)
Requirement already satisfied: pycryptodome>=3.8.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from bce-python-sdk->visualdl>=2.2.0->paddleclas>=2.4.3) (3.19.0)
Requirement already satisfied: fonttools>=4.22.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from matplotlib->visualdl>=2.2.0->paddleclas>=2.4.3) (4.25.0)
Requirement already satisfied: contourpy>=1.0.1 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from matplotlib->visualdl>=2.2.0->paddleclas>=2.4.3) (1.0.5)
Requirement already satisfied: python-dateutil>=2.7 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from matplotlib->visualdl>=2.2.0->paddleclas>=2.4.3) (2.8.2)
Requirement already satisfied: kiwisolver>=1.0.1 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from matplotlib->visualdl>=2.2.0->paddleclas>=2.4.3) (1.4.4)
Requirement already satisfied: cycler>=0.10 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from matplotlib->visualdl>=2.2.0->paddleclas>=2.4.3) (0.11.0)
Requirement already satisfied: pyparsing>=2.3.1 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from matplotlib->visualdl>=2.2.0->paddleclas>=2.4.3) (3.1.0)
Requirement already satisfied: idna<4,>=2.5 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from requests->visualdl>=2.2.0->paddleclas>=2.4.3) (3.4)
Requirement already satisfied: urllib3<3,>=1.21.1 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from requests->visualdl>=2.2.0->paddleclas>=2.4.3) (1.26.14)
Requirement already satisfied: certifi>=2017.4.17 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from requests->visualdl>=2.2.0->paddleclas>=2.4.3) (2023.5.7)
Requirement already satisfied: charset-normalizer<4,>=2 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from requests->visualdl>=2.2.0->paddleclas>=2.4.3) (3.2.0)
Requirement already satisfied: MarkupSafe>=2.0 in /Users/zhaohua/anaconda3/lib/python3.10/site-packages (from Jinja2>=3.0->flask>=1.1.1->visualdl>=2.2.0->paddleclas>=2.4.3) (2.1.1)
Building wheels for collected packages: faiss-cpu
  Building wheel for faiss-cpu (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [85 lines of output]
      /Users/zhaohua/anaconda3/lib/python3.10/site-packages/setuptools/__init__.py:84: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
      !!

              ********************************************************************************
              Requirements should be satisfied by a PEP 517 installer.
              If you are using pip, you can try `pip install --use-pep517`.
              ********************************************************************************

      !!
        dist.fetch_build_eggs(dist.setup_requires)
      running bdist_wheel
      running build
      running build_py
      running build_ext
      building 'faiss._swigfaiss' extension
      swigging faiss/faiss/python/swigfaiss.i to faiss/faiss/python/swigfaiss_wrap.cpp
      swig -python -c++ -Doverride= -I/usr/local/include -Ifaiss -o faiss/faiss/python/swigfaiss_wrap.cpp faiss/faiss/python/swigfaiss.i
      faiss/faiss/python/swigfaiss.i:250: Error: Unable to find 'faiss/impl/platform_macros.h'
      faiss/faiss/python/swigfaiss.i:254: Error: Unable to find 'faiss/utils/ordered_key_value.h'
      faiss/faiss/python/swigfaiss.i:255: Error: Unable to find 'faiss/utils/Heap.h'
      faiss/faiss/python/swigfaiss.i:260: Error: Unable to find 'faiss/utils/AlignedTable.h'
      faiss/faiss/python/swigfaiss.i:261: Error: Unable to find 'faiss/utils/partitioning.h'
      faiss/faiss/python/swigfaiss.i:262: Error: Unable to find 'faiss/utils/hamming.h'
      faiss/faiss/python/swigfaiss.i:365: Error: Unable to find 'faiss/utils/utils.h'
      faiss/faiss/python/swigfaiss.i:366: Error: Unable to find 'faiss/utils/distances.h'
      faiss/faiss/python/swigfaiss.i:367: Error: Unable to find 'faiss/utils/random.h'
      faiss/faiss/python/swigfaiss.i:369: Error: Unable to find 'faiss/MetricType.h'
      faiss/faiss/python/swigfaiss.i:372: Error: Unable to find 'faiss/Index.h'
      faiss/faiss/python/swigfaiss.i:373: Error: Unable to find 'faiss/Clustering.h'
      faiss/faiss/python/swigfaiss.i:375: Error: Unable to find 'faiss/utils/extra_distances.h'
      faiss/faiss/python/swigfaiss.i:379: Error: Unable to find 'faiss/impl/ProductQuantizer.h'
      faiss/faiss/python/swigfaiss.i:380: Error: Unable to find 'faiss/impl/AdditiveQuantizer.h'
      faiss/faiss/python/swigfaiss.i:381: Error: Unable to find 'faiss/impl/ResidualQuantizer.h'
      faiss/faiss/python/swigfaiss.i:382: Error: Unable to find 'faiss/impl/LocalSearchQuantizer.h'
      faiss/faiss/python/swigfaiss.i:384: Error: Unable to find 'faiss/VectorTransform.h'
      faiss/faiss/python/swigfaiss.i:385: Error: Unable to find 'faiss/IndexPreTransform.h'
      faiss/faiss/python/swigfaiss.i:386: Error: Unable to find 'faiss/IndexFlat.h'
      faiss/faiss/python/swigfaiss.i:387: Error: Unable to find 'faiss/IndexRefine.h'
      faiss/faiss/python/swigfaiss.i:388: Error: Unable to find 'faiss/IndexLSH.h'
      faiss/faiss/python/swigfaiss.i:389: Error: Unable to find 'faiss/impl/PolysemousTraining.h'
      faiss/faiss/python/swigfaiss.i:390: Error: Unable to find 'faiss/IndexPQ.h'
      faiss/faiss/python/swigfaiss.i:391: Error: Unable to find 'faiss/IndexResidual.h'
      faiss/faiss/python/swigfaiss.i:392: Error: Unable to find 'faiss/impl/io.h'
      faiss/faiss/python/swigfaiss.i:394: Error: Unable to find 'faiss/invlists/InvertedLists.h'
      faiss/faiss/python/swigfaiss.i:395: Error: Unable to find 'faiss/invlists/InvertedListsIOHook.h'
      faiss/faiss/python/swigfaiss.i:397: Error: Unable to find 'faiss/invlists/BlockInvertedLists.h'
      faiss/faiss/python/swigfaiss.i:398: Error: Unable to find 'faiss/invlists/DirectMap.h'
      faiss/faiss/python/swigfaiss.i:401: Error: Unable to find 'faiss/IndexIVF.h'
      faiss/faiss/python/swigfaiss.i:406: Error: Unable to find 'faiss/IVFlib.h'
      faiss/faiss/python/swigfaiss.i:407: Error: Unable to find 'faiss/impl/ScalarQuantizer.h'
      faiss/faiss/python/swigfaiss.i:408: Error: Unable to find 'faiss/IndexScalarQuantizer.h'
      faiss/faiss/python/swigfaiss.i:409: Error: Unable to find 'faiss/IndexIVFSpectralHash.h'
      faiss/faiss/python/swigfaiss.i:410: Error: Unable to find 'faiss/impl/HNSW.h'
      faiss/faiss/python/swigfaiss.i:411: Error: Unable to find 'faiss/IndexHNSW.h'
      faiss/faiss/python/swigfaiss.i:414: Error: Unable to find 'faiss/impl/NNDescent.h'
      faiss/faiss/python/swigfaiss.i:415: Error: Unable to find 'faiss/IndexNNDescent.h'
      faiss/faiss/python/swigfaiss.i:417: Error: Unable to find 'faiss/IndexIVFFlat.h'
      faiss/faiss/python/swigfaiss.i:418: Error: Unable to find 'faiss/impl/NSG.h'
      faiss/faiss/python/swigfaiss.i:419: Error: Unable to find 'faiss/IndexNSG.h'
      faiss/faiss/python/swigfaiss.i:424: Error: Unable to find 'faiss/invlists/OnDiskInvertedLists.h'
      faiss/faiss/python/swigfaiss.i:427: Error: Unable to find 'faiss/impl/lattice_Zn.h'
      faiss/faiss/python/swigfaiss.i:428: Error: Unable to find 'faiss/IndexLattice.h'
      faiss/faiss/python/swigfaiss.i:431: Error: Unable to find 'faiss/IndexIVFPQ.h'
      faiss/faiss/python/swigfaiss.i:432: Error: Unable to find 'faiss/IndexIVFPQR.h'
      faiss/faiss/python/swigfaiss.i:433: Error: Unable to find 'faiss/Index2Layer.h'
      faiss/faiss/python/swigfaiss.i:435: Error: Unable to find 'faiss/IndexPQFastScan.h'
      faiss/faiss/python/swigfaiss.i:436: Error: Unable to find 'faiss/IndexIVFPQFastScan.h'
      faiss/faiss/python/swigfaiss.i:437: Error: Unable to find 'faiss/utils/quantize_lut.h'
      faiss/faiss/python/swigfaiss.i:439: Error: Unable to find 'faiss/IndexBinary.h'
      faiss/faiss/python/swigfaiss.i:440: Error: Unable to find 'faiss/IndexBinaryFlat.h'
      faiss/faiss/python/swigfaiss.i:441: Error: Unable to find 'faiss/IndexBinaryIVF.h'
      faiss/faiss/python/swigfaiss.i:442: Error: Unable to find 'faiss/IndexBinaryFromFloat.h'
      faiss/faiss/python/swigfaiss.i:443: Error: Unable to find 'faiss/IndexBinaryHNSW.h'
      faiss/faiss/python/swigfaiss.i:444: Error: Unable to find 'faiss/IndexBinaryHash.h'
      faiss/faiss/python/swigfaiss.i:446: Error: Unable to find 'faiss/impl/ThreadedIndex.h'
      faiss/faiss/python/swigfaiss.i:450: Error: Unable to find 'faiss/IndexShards.h'
      faiss/faiss/python/swigfaiss.i:454: Error: Unable to find 'faiss/IndexReplicas.h'
      faiss/faiss/python/swigfaiss.i:458: Error: Unable to find 'faiss/MetaIndexes.h'
      faiss/faiss/python/swigfaiss.i:473: Error: Unable to find 'faiss/impl/AuxIndexStructures.h'
      faiss/faiss/python/swigfaiss.i:663: Error: Unable to find 'faiss/index_io.h'
      faiss/faiss/python/swigfaiss.i:664: Error: Unable to find 'faiss/clone_index.h'
      faiss/faiss/python/swigfaiss.i:668: Error: Unable to find 'faiss/AutoTune.h'
      faiss/faiss/python/swigfaiss.i:669: Error: Unable to find 'faiss/index_factory.h'
      faiss/faiss/python/swigfaiss.i:670: Error: Unable to find 'faiss/MatrixStats.h'
      error: command '/opt/homebrew/bin/swig' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for faiss-cpu
  Running setup.py clean for faiss-cpu
Failed to build faiss-cpu
Installing collected packages: faiss-cpu, easydict, prettytable, gast, paddleclas
  Attempting uninstall: faiss-cpu
    Found existing installation: faiss-cpu 1.7.3
    Uninstalling faiss-cpu-1.7.3:
      Successfully uninstalled faiss-cpu-1.7.3
  Running setup.py install for faiss-cpu ... error
  error: subprocess-exited-with-error

  × Running setup.py install for faiss-cpu did not run successfully.
  │ exit code: 1
  ╰─> [98 lines of output]
      /Users/zhaohua/anaconda3/lib/python3.10/site-packages/setuptools/__init__.py:84: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
      !!

              ********************************************************************************
              Requirements should be satisfied by a PEP 517 installer.
              If you are using pip, you can try `pip install --use-pep517`.
              ********************************************************************************

      !!
        dist.fetch_build_eggs(dist.setup_requires)
      running install
      /Users/zhaohua/anaconda3/lib/python3.10/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
      !!

              ********************************************************************************
              Please avoid running ``setup.py`` directly.
              Instead, use pypa/build, pypa/installer or other
              standards-based tools.

              See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
              ********************************************************************************

      !!
        self.initialize_options()
      running build
      running build_py
      running build_ext
      building 'faiss._swigfaiss' extension
      swigging faiss/faiss/python/swigfaiss.i to faiss/faiss/python/swigfaiss_wrap.cpp
      swig -python -c++ -Doverride= -I/usr/local/include -Ifaiss -o faiss/faiss/python/swigfaiss_wrap.cpp faiss/faiss/python/swigfaiss.i
      faiss/faiss/python/swigfaiss.i:250: Error: Unable to find 'faiss/impl/platform_macros.h'
      faiss/faiss/python/swigfaiss.i:254: Error: Unable to find 'faiss/utils/ordered_key_value.h'
      faiss/faiss/python/swigfaiss.i:255: Error: Unable to find 'faiss/utils/Heap.h'
      faiss/faiss/python/swigfaiss.i:260: Error: Unable to find 'faiss/utils/AlignedTable.h'
      faiss/faiss/python/swigfaiss.i:261: Error: Unable to find 'faiss/utils/partitioning.h'
      faiss/faiss/python/swigfaiss.i:262: Error: Unable to find 'faiss/utils/hamming.h'
      faiss/faiss/python/swigfaiss.i:365: Error: Unable to find 'faiss/utils/utils.h'
      faiss/faiss/python/swigfaiss.i:366: Error: Unable to find 'faiss/utils/distances.h'
      faiss/faiss/python/swigfaiss.i:367: Error: Unable to find 'faiss/utils/random.h'
      faiss/faiss/python/swigfaiss.i:369: Error: Unable to find 'faiss/MetricType.h'
      faiss/faiss/python/swigfaiss.i:372: Error: Unable to find 'faiss/Index.h'
      faiss/faiss/python/swigfaiss.i:373: Error: Unable to find 'faiss/Clustering.h'
      faiss/faiss/python/swigfaiss.i:375: Error: Unable to find 'faiss/utils/extra_distances.h'
      faiss/faiss/python/swigfaiss.i:379: Error: Unable to find 'faiss/impl/ProductQuantizer.h'
      faiss/faiss/python/swigfaiss.i:380: Error: Unable to find 'faiss/impl/AdditiveQuantizer.h'
      faiss/faiss/python/swigfaiss.i:381: Error: Unable to find 'faiss/impl/ResidualQuantizer.h'
      faiss/faiss/python/swigfaiss.i:382: Error: Unable to find 'faiss/impl/LocalSearchQuantizer.h'
      faiss/faiss/python/swigfaiss.i:384: Error: Unable to find 'faiss/VectorTransform.h'
      faiss/faiss/python/swigfaiss.i:385: Error: Unable to find 'faiss/IndexPreTransform.h'
      faiss/faiss/python/swigfaiss.i:386: Error: Unable to find 'faiss/IndexFlat.h'
      faiss/faiss/python/swigfaiss.i:387: Error: Unable to find 'faiss/IndexRefine.h'
      faiss/faiss/python/swigfaiss.i:388: Error: Unable to find 'faiss/IndexLSH.h'
      faiss/faiss/python/swigfaiss.i:389: Error: Unable to find 'faiss/impl/PolysemousTraining.h'
      faiss/faiss/python/swigfaiss.i:390: Error: Unable to find 'faiss/IndexPQ.h'
      faiss/faiss/python/swigfaiss.i:391: Error: Unable to find 'faiss/IndexResidual.h'
      faiss/faiss/python/swigfaiss.i:392: Error: Unable to find 'faiss/impl/io.h'
      faiss/faiss/python/swigfaiss.i:394: Error: Unable to find 'faiss/invlists/InvertedLists.h'
      faiss/faiss/python/swigfaiss.i:395: Error: Unable to find 'faiss/invlists/InvertedListsIOHook.h'
      faiss/faiss/python/swigfaiss.i:397: Error: Unable to find 'faiss/invlists/BlockInvertedLists.h'
      faiss/faiss/python/swigfaiss.i:398: Error: Unable to find 'faiss/invlists/DirectMap.h'
      faiss/faiss/python/swigfaiss.i:401: Error: Unable to find 'faiss/IndexIVF.h'
      faiss/faiss/python/swigfaiss.i:406: Error: Unable to find 'faiss/IVFlib.h'
      faiss/faiss/python/swigfaiss.i:407: Error: Unable to find 'faiss/impl/ScalarQuantizer.h'
      faiss/faiss/python/swigfaiss.i:408: Error: Unable to find 'faiss/IndexScalarQuantizer.h'
      faiss/faiss/python/swigfaiss.i:409: Error: Unable to find 'faiss/IndexIVFSpectralHash.h'
      faiss/faiss/python/swigfaiss.i:410: Error: Unable to find 'faiss/impl/HNSW.h'
      faiss/faiss/python/swigfaiss.i:411: Error: Unable to find 'faiss/IndexHNSW.h'
      faiss/faiss/python/swigfaiss.i:414: Error: Unable to find 'faiss/impl/NNDescent.h'
      faiss/faiss/python/swigfaiss.i:415: Error: Unable to find 'faiss/IndexNNDescent.h'
      faiss/faiss/python/swigfaiss.i:417: Error: Unable to find 'faiss/IndexIVFFlat.h'
      faiss/faiss/python/swigfaiss.i:418: Error: Unable to find 'faiss/impl/NSG.h'
      faiss/faiss/python/swigfaiss.i:419: Error: Unable to find 'faiss/IndexNSG.h'
      faiss/faiss/python/swigfaiss.i:424: Error: Unable to find 'faiss/invlists/OnDiskInvertedLists.h'
      faiss/faiss/python/swigfaiss.i:427: Error: Unable to find 'faiss/impl/lattice_Zn.h'
      faiss/faiss/python/swigfaiss.i:428: Error: Unable to find 'faiss/IndexLattice.h'
      faiss/faiss/python/swigfaiss.i:431: Error: Unable to find 'faiss/IndexIVFPQ.h'
      faiss/faiss/python/swigfaiss.i:432: Error: Unable to find 'faiss/IndexIVFPQR.h'
      faiss/faiss/python/swigfaiss.i:433: Error: Unable to find 'faiss/Index2Layer.h'
      faiss/faiss/python/swigfaiss.i:435: Error: Unable to find 'faiss/IndexPQFastScan.h'
      faiss/faiss/python/swigfaiss.i:436: Error: Unable to find 'faiss/IndexIVFPQFastScan.h'
      faiss/faiss/python/swigfaiss.i:437: Error: Unable to find 'faiss/utils/quantize_lut.h'
      faiss/faiss/python/swigfaiss.i:439: Error: Unable to find 'faiss/IndexBinary.h'
      faiss/faiss/python/swigfaiss.i:440: Error: Unable to find 'faiss/IndexBinaryFlat.h'
      faiss/faiss/python/swigfaiss.i:441: Error: Unable to find 'faiss/IndexBinaryIVF.h'
      faiss/faiss/python/swigfaiss.i:442: Error: Unable to find 'faiss/IndexBinaryFromFloat.h'
      faiss/faiss/python/swigfaiss.i:443: Error: Unable to find 'faiss/IndexBinaryHNSW.h'
      faiss/faiss/python/swigfaiss.i:444: Error: Unable to find 'faiss/IndexBinaryHash.h'
      faiss/faiss/python/swigfaiss.i:446: Error: Unable to find 'faiss/impl/ThreadedIndex.h'
      faiss/faiss/python/swigfaiss.i:450: Error: Unable to find 'faiss/IndexShards.h'
      faiss/faiss/python/swigfaiss.i:454: Error: Unable to find 'faiss/IndexReplicas.h'
      faiss/faiss/python/swigfaiss.i:458: Error: Unable to find 'faiss/MetaIndexes.h'
      faiss/faiss/python/swigfaiss.i:473: Error: Unable to find 'faiss/impl/AuxIndexStructures.h'
      faiss/faiss/python/swigfaiss.i:663: Error: Unable to find 'faiss/index_io.h'
      faiss/faiss/python/swigfaiss.i:664: Error: Unable to find 'faiss/clone_index.h'
      faiss/faiss/python/swigfaiss.i:668: Error: Unable to find 'faiss/AutoTune.h'
      faiss/faiss/python/swigfaiss.i:669: Error: Unable to find 'faiss/index_factory.h'
      faiss/faiss/python/swigfaiss.i:670: Error: Unable to find 'faiss/MatrixStats.h'
      error: command '/opt/homebrew/bin/swig' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  Rolling back uninstall of faiss-cpu
  Moving to /Users/zhaohua/anaconda3/lib/python3.10/site-packages/faiss/
   from /Users/zhaohua/anaconda3/lib/python3.10/site-packages/~aiss
  Moving to /Users/zhaohua/anaconda3/lib/python3.10/site-packages/faiss_cpu-1.7.3.dist-info/
   from /Users/zhaohua/anaconda3/lib/python3.10/site-packages/~aiss_cpu-1.7.3.dist-info
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> faiss-cpu

<img width="280" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/3362757/45b0ae6f-e2a8-4423-a121-a22c5f80faf2">
<img width="388" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/3362757/31db034c-d142-449a-888a-a67c63a3d290">

**Comment by zhh8689** [comment]
没有人回答吗？

**Comment by SWHL** [comment]
看着问题是faiss-cpu安装有问题
